BlogTemplate.Tests/Pages/EditTests.cs

```diff
@@ -43,5 +43,101 @@ public void UpdatePostP2P_TitleIsUpdated_UpdateSlug()
 
             Assert.Equal("Edited-Title", post.Slug);
         }
+        [Fact]
+        public void UpdatePostD2P_TitleIsUpdated_UpdateSlug()
+        {
+            IFileSystem testFileSystem = new FakeFileSystem();
+            BlogDataStore testDataStore = new BlogDataStore(new FakeFileSystem());
+            SlugGenerator slugGenerator = new SlugGenerator(testDataStore);
+            ExcerptGenerator excerptGenerator = new ExcerptGenerator();
+
+            Post post = new Post()
+            {
+                Title = "Title",
+                Slug = "Title",
+                Body = "Body",
+                Excerpt = "Excerpt",
+                IsPublic = false,
+            };
+
+            testDataStore.SavePost(post);
+
+            EditModel testEditModel = new EditModel(testDataStore, slugGenerator, excerptGenerator);
+            testEditModel.PageContext = new PageContext();
+            testEditModel.OnGet(post.Id);
+            testEditModel.EditedPost.Title = "Edited Title";
+            testEditModel.OnPostPublish(post.Id, true);
+
+            post = testDataStore.GetPost(post.Id);
+
+            Assert.Equal("Edited-Title", post.Slug);
+        }
+        [Fact]
+        public void UpdatePostP2D2P_TitleIsUpdated_UpdateSlug()

```

Comment:
Expand P2D2P and name appropriately

Reply:
For that matter P2D2P is actually two tests: P2D and D2P.  As it stands now, if P2D fails, you won't get test coverage on D2P, unless you have another test covering that.  And if you do, then this test is redundant.

A better way to set up the test would be to do P2D as one test, then D2P may have multiple tests:

1. D2P where the Draft has never been published (no PubDate, no Comments, etc)
2. D2P where the Draft has been previously published (PubDate set, maybe have some Comments, etc)